src/uct/gaudi/base/gaudi_base.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) Intel Corporation, 2025. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "gaudi_base.h"
+#include <uct/gaudi/gaudi_gdr/gaudi_gdr_md.h>
+#include <ucs/sys/module.h>
+#include <ucs/memory/numa.h>
+#include <ucs/sys/sys.h>
+#include <ucs/sys/topo/base/topo.h>
+#include <pthread.h>
+
+#include <inttypes.h>
+#include <fcntl.h>
+#include <hlthunk.h>
+#include <synapse_api.h>
+
+int uct_gaudi_base_get_fd(int device_id, bool *fd_created) {
+    synDeviceInfo deviceInfo;
+
+    if (synDeviceGetInfo(-1, &deviceInfo) != synSuccess) {
+        int fd = hlthunk_open_by_module_id(device_id);
+        if (fd <0) {
+            ucs_info("Failed to get device fd via hlthunk_open_by_module_id, id %d", device_id);
+	         fd = hlthunk_open(HLTHUNK_DEVICE_DONT_CARE, NULL);
+        }
+
+        if (fd >=0 && fd_created != NULL) {
+            *fd_created = true;
+        }
+        return fd;
+    }
+
+    return deviceInfo.fd;
+}
+
+void uct_gaudi_base_close_fd(int fd, bool fd_created) {
+    if (fd_created && fd >= 0) {
+        hlthunk_close(fd);
+    }
+}
+
+void uct_gaudi_base_close_dmabuf_fd(int fd) {
+    if (fd >= 0) {
+        close(fd);
+    }
+}
+
+ucs_status_t uct_gaudi_base_get_sysdev(int fd, ucs_sys_device_t* sys_dev) {
+    ucs_status_t status;
+    char pci_bus_id[13];
+    int rc = hlthunk_get_pci_bus_id_from_fd(fd, pci_bus_id, sizeof(pci_bus_id));
```

Comment:
pls don't use c99 style declarations (even in technically it compiles)


```suggestion
    int rc;
    
    rc = hlthunk_get_pci_bus_id_from_fd(fd, pci_bus_id, sizeof(pci_bus_id));
```